File: software_1/101.c
#include <stdint.h>
#include <stdbool.h>
#include <stdio.h>

/* WARNING: Function: __x86.get_pc_thunk.bx replaced with injection: get_pc_thunk_bx */

int main(void)
{
  uint32_t *arr_ptr;
  uint32_t arr [6];

  arr[0] = 0xabb6bcbc;
  arr[1] = 0x9d9b9884;
  arr[2] = 0xa0cf8ba0;
  arr[3] = 0xa0cc978b;
  arr[4] = 0x9cca9a8d;
  arr[5] = 0xff829a8a;
  arr_ptr = arr;
  while( true ) {
    if ((char)~*(char *)arr_ptr == 0) break;
    putchar(~*(char *)arr_ptr);
    arr_ptr = (uint32_t *)((int)arr_ptr + 1);
  }
  putchar(L'\n');
  return 0;
}

File: software_1/101.py

#integers found in function print_safe_contents
arr = [0xabb6bcbc, 0x9d9b9884, 0xa0cf8ba0, 0xa0cc978b, 0x9cca9a8d, 0xff829a8a]

#interpret longs as bytes in little endian
arr = [num.to_bytes(4, byteorder='little') for num in arr]

# flatten nested list into single list
arr = [ char for b_arr in arr for char in b_arr]

# Apply transformation seen in print_safe_contents
chrs = [chr((~i) & 0xFF) for i in arr]
result = ''.join(chrs)
print(result)
